plugins/parsers/registry.go

```diff
@@ -139,12 +150,49 @@ func NewParser(config *Config) (Parser, error) {
 			config.GrokCustomPatterns,
 			config.GrokCustomPatternFiles,
 			config.GrokTimeZone)
+	case "csv":
+		parser, err = newCSVParser(config.MetricName,
+			config.CSVHeader,
+			config.CSVDelimiter,
+			config.CSVDataColumns,
+			config.CSVTagColumns,
+			config.CSVFieldColumns,
+			config.CSVNameColumn,
+			config.CSVTimestampColumn,
+			config.CSVTimestampFormat,
+			config.DefaultTags)
 	default:
 		err = fmt.Errorf("Invalid data format: %s", config.DataFormat)
 	}
 	return parser, err
 }
 
+func newCSVParser(metricName string,
+	header bool,
+	delimiter string,
+	dataColumns []string,
+	tagColumns []string,
+	fieldColumns []string,
+	nameColumn string,
+	timestampColumn string,
+	timestampFormat string,
+	defaultTags map[string]string) (Parser, error) {
```

Comment:
Returning an error isn't useful here. Where it's an unexported function not matching any interface, I'd remove it, or remove the function altogether and just instantiate a `CSVParser` on line 154.